File: solutions/2455. Average Value of Even Numbers That Are Divisible by Three/2455.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int averageValue(const vector<int>& nums) {
    int count = 0;
    int sum = 0;

    for (const int num : nums)
      if (num % 6 == 0) {
        sum += num;
        ++count;
      }

    return count == 0 ? 0 : sum / count;
  }
};

File: solutions/2455. Average Value of Even Numbers That Are Divisible by Three/2455.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def averageValue(self, nums: list[int]) -> int:
    count = 0
    summ = 0

    for num in nums:
      if num % 6 == 0:
        summ += num
        count += 1

    return 0 if count == 0 else summ // count